### PR TITLE
Update to Jupyterlab 3.0

### LINKED ISSUE #779: Update for jupyterlab 3
**edit - Install jupyterlab-git beta**: Jupyterlab-git has a beta release which you can install with:
```
pip install --upgrade --pre jupyterlab-git
```

## Description
Jupyterlab version 3 is nearly upon us (rc this week I think?). 
The update to jlab3 may be as simple as running: `python -m jupyterlab.upgrade_extension .` but what should be done w.r.t branches on this repo? Should master track jlab 3 and then create a `0.2x.y` branch for bugfixes for jlab2?

### PR DESCRIPTION
**Pull request by ajbozarth**
Update to support Jupyterlab 3.0 and addresses any issues accosted with that update.

Currently in draft until final release when this will be updated.

This PR is paired with https://github.com/jupyter/nbdime/pull/551 which is require to test this, including a temporary 1 line hard-coded local import of nbdime 

Fixes #779

### CODE REVIEW

**Comment by ajbozarth** [comment]
note: CI will fail until nbdime PR is merged and released

File: jupyterlab_git/_version.py
Comment (fcollonval) [inline jupyterlab_git/_version.py:4]:
```suggestion
version_info = (0, 30, 0)
```

File: package.json
Comment (fcollonval) [inline package.json:3]:
```suggestion
  "version": "0.30.0",
```

File: package.json
Comment (fcollonval) [inline package.json:76]: Just to highlight this needs to be updated once nbdime has been updated.

**Review by fcollonval — COMMENTED** [review]
Thanks for starting this @ajbozarth 

I push some suggestions to correct the version number to match the current policy.

**Comment by ianhi** [comment]
I think we should also turn this into a federated extension to eliminate the build step on installation. Thoughts on this?

File: jupyterlab_git/_version.py
Comment (ajbozarth) [reply jupyterlab_git/_version.py:4]: why jump from 22 to 30?

**Comment by ajbozarth** [comment]
> I think we should also turn this into a federated extension 

I agree, but I believe that initially just updating to 3.0 as a normal extension to be release same day as jupyterlab 3.0 then following up with an update to a federated extension later is the safer option and will cause less delays to users who want to switch to 3.0

File: jupyterlab_git/_version.py
Comment (fcollonval) [reply jupyterlab_git/_version.py:4]: When the extension was ported to JupyterLab 2, it was decided to jump from 0.10 to 0.20 to left version available for version with backported fix and features. So it will be good to do the same with JLab 3 to left version 0.2x available for backports for JupyterLab 2.

File: setup.py
Comment (ajbozarth) [inline setup.py:83]: also leaving myself a note this will need to be updated when nbdime is updated as well

**Comment by ajbozarth** [comment]
recently update the nbdime PR and requested a rc, once an nbdime rc is released I'll update this PR

**Comment by ajbozarth** [comment]
I've pushed some updates so we can get some review started on this before the nbdime pr is published. Currently the tests are broken and I'm open to others pushing to this PR to address them if you have ideas.

As for checking out and running this PR, you will need to also checkout and build the linked nbdime PR and update the package.json here to the relative location of that project on your machine for this to work.

**Comment by ajbozarth** [comment]
just pushed an update to use the nbdime beta, this PR should now be usable without needing to install a local dev instance of nbdime

**Comment by fcollonval** [comment]
Thanks @ajbozarth 

We are almost there. There is only one fix to figure out why looking into the React tree is not working anymore. Happy to get some help on that one.

File: setup.py
Comment (fcollonval) [inline setup.py:69]:
```suggestion
    install_requires=["jupyter_server", "nbdime~=3.0.0b1", "packaging", "pexpect"],
```

`jupyter_packaging` is not a install dependency but a setup one (only need by dev'). We should add it to the extra requires - and probably rename it as `dev` rather than `test` as there are already other packages like pre-commit or black in there.

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/git-extension-for-jupyterlab-3/7569/2

**Comment by ajbozarth** [comment]
@fcollonval @lresende thank you both for helping on this, is this good to merge now? If not what's left blocking it and releasing an rc or beta?

**Comment by fcollonval** [comment]
> @fcollonval @lresende thank you both for helping on this, is this good to merge now? If not what's left blocking it and releasing an rc or beta?

I'm ok to release a beta. For my information, on the `nbdime` front, what is missing to get a release?

**Review by fcollonval — APPROVED** [review]

**Pull request merged**
